api/v1alpha1/utils/vaultutils.go

```diff
@@ -40,7 +48,7 @@ func read(context context.Context, path string) (map[string]interface{}, bool, e
 	secret, err := vaultClient.Logical().Read(path)
 	if err != nil {
 		if respErr, ok := err.(*vault.ResponseError); ok {
-			if respErr.StatusCode == 404 {
+			if respErr.StatusCode == 404 || respErr.StatusCode == 204 {
```

Comment:
204 is No Content. I will use it to check if the CA is already generated `/pki/ca/pem` in case there is a misalignment between the vault and the CR, like manual deletion of the root CA.

Reply:
mmm, this will change the behavior for everyone correct? I'm not sure it's right.